k8s-operator/sessionrecording/hijacker.go

```diff
@@ -19,17 +21,51 @@ import (
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"tailscale.com/client/tailscale/apitype"
+	"tailscale.com/k8s-operator/sessionrecording/spdy"
+	"tailscale.com/k8s-operator/sessionrecording/tsrecorder"
+	"tailscale.com/sessionrecording"
 	"tailscale.com/tailcfg"
 	"tailscale.com/tsnet"
 	"tailscale.com/tstime"
+	"tailscale.com/util/clientmetric"
 	"tailscale.com/util/multierr"
 )
 
-// spdyHijacker implements [net/http.Hijacker] interface.
+const SPDYProtocol protocol = "SPDY"
+
+var (
+	// counterSessionRecordingsAttempted counts the number of session recording attempts.
+	CounterSessionRecordingsAttempted = clientmetric.NewCounter("k8s_auth_proxy__session_recordings_attempted")
+
+	// counterSessionRecordingsUploaded counts the number of successfully uploaded session recordings.
+	CounterSessionRecordingsUploaded = clientmetric.NewCounter("k8s_auth_proxy_session_recordings_uploaded")
```

Comment:
Separate to the above comments, do either of these need to be exported?

I can see why the "attempts" metric needs to be accessible from outside the package, but maybe it would be a cleaner package interface to only expose a function that can increment the metric? As it is it feels like we're exposing more than we need to, especially as they're mutable variables.

Reply:
it's a bit annoying, one of them is only accessed from outside and one only from inside as that is tied to the connection setup/failure/success processing flows. I want to keep them together and looking the same as the other metrics vars are set up to make it easier to discover what metrics we are collecting. I made the metric the metric that is only accessed from inside private. I don't see much point in a function for the other one- it is only modified by a single external actor in the only one way it can be modified and nothing else should ever need to access it for anything.